naming/src/test/java/org/jboss/as/naming/subsystem/NamingSubsystemTestCase.java

```diff
@@ -123,6 +125,34 @@ public void testCompositeBindingUpdate() throws Exception {
         ModelTestUtils.checkOutcome(services.executeOperation(compositeOp));
     }
 
+    @Test
+    public void testExpressionInAttributeValue() throws Exception{
+        final String subsystemXml = readResource("subsystem_expression.xml");
+        final KernelServices ks = createKernelServicesBuilder(createAdditionalInitialization()).setSubsystemXml(subsystemXml).build();
+        final ModelNode naming = ks.readWholeModel().get("subsystem", getMainSubsystemName());
+         ModelNode attribute = naming.get(NamingSubsystemModel.BINDING).get("java:global/a");
+
+        final String value = attribute.get(NamingSubsystemModel.VALUE).resolve().asString();
```

Comment:
Calling ModelNode.resolve is more of a test of JBoss DMR, which is tested there.

To confirm that the subsystem code is handling the expressions correctly, instead create ModelNode that represent a /subsystem=naming:read-resource(recursive=true,resolve-expressions=true) operation and use KernelService.executeOperation to run it. The result should have resolved values and you can directly check them.